pkg/resource/certificate_authority_activation/hooks.go

```diff
@@ -193,6 +196,61 @@ func (rm *resourceManager) customUpdateCertificateAuthorityActivation(
 	return &resource{ko}, nil
 }
 
+func (rm *resourceManager) writeCertificateChainToSecret(
+	ctx context.Context,
+	certificate string,
+	certificateChain string,
+	r *resource,
+) (err error) {
```

Comment:
Maybe we just pass the annotations object here?
```suggestion
func (rm *resourceManager) writeCertificateChainToSecret(
	ctx context.Context,
	annotations *metav1.ObjectMeta,
	certificate string,
	certificateChain string,
```